### PR TITLE
Add CAPX Quickstart test as a CCM E2E Test

### PR DESCRIPTION
**Pull request by thunderboltsid**
This PR sets up the initial E2E infrastructure to run E2E Presubmits on CCM and adds the CAPX quickstart test as the first test. In future, we'd want to add cluster topology test (multi-pe) as part of e2e testing as well as CCM specific e2e tests.

**How has this been tested?**
Add a ProwJob to run the e2e test added in this PR, run the prow job, and see it pass.

### CODE REVIEW

**Review by deepakm-ntnx — APPROVED** [review]
Does upstream cloud-provider has any e2e inbuilt as well which we can run here?

File: Makefile [hunk 1]
```diff
@@ -8,33 +8,35 @@ IMG_TAG ?= latest
 
 EXPORT_RESULT?=false # for CI please set EXPORT_RESULT to true
 
+GOCMD=go
```
Comment (deepakm-ntnx) [inline Makefile:11]: next thing will be to use devbox

File: test/e2e/capx_test.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	"sigs.k8s.io/cluster-api/test/e2e"
+)
+
+var _ = Describe("CCM on a CAPX Cluster", Label("capx"), func() {
+	e2e.QuickStartSpec(ctx, func() e2e.QuickStartSpecInput {
```
Comment (deepakm-ntnx) [inline test/e2e/capx_test.go:11]: another option is to use following code and also verify if ccm is installed <pre>
By("Creating a workload cluster (no additional categories)", func() {
			testHelper.deployClusterAndWait(
				deployClusterParams{
					clusterName:           clusterName,
					namespace:             namespace,
					flavor:                flavor,
					clusterctlConfigPath:  clusterctlConfigPath,
					artifactFolder:        artifactFolder,
					bootstrapClusterProxy: bootstrapClusterProxy,
				}, clusterResources)
		})

		By("Checking cluster category condition is true", func() {
			testHelper.verifyConditionOnNutanixCluster(verifyConditionParams{
				clusterName:           clusterName,
				namespace:             namespace,
				bootstrapClusterProxy: bootstrapClusterProxy,
				expectedCondition: clusterv1.Condition{
					Type:   infrav1.ClusterCategoryCreatedCondition,
					Status: corev1.ConditionTrue,
				},
			})
		})
</pre>

Comment (deepakm-ntnx) [reply test/e2e/capx_test.go:11]: since quickstart wont check if ccm got installed or not

Comment (deepakm-ntnx) [reply test/e2e/capx_test.go:11]: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/main/test/e2e/basic_uuid_test.go

Comment (deepakm-ntnx) [reply test/e2e/capx_test.go:11]: we will also need one with clusterclass flavor as well then

Comment (thunderboltsid) [reply test/e2e/capx_test.go:11]: QuickStart will fail if CCM is not installed. So test passing actually ensures that CCM is installed and doing its job.

File: Makefile (hunk 1 above)
Comment (thunderboltsid) [reply Makefile:11]: > next thing will be to use devbox

Yeah, @faiq is working with Security and Legal so we can get the jetpack devbox install GitHub action allowed.

**Comment by thunderboltsid** [comment]
> Does upstream cloud-provider has any e2e inbuilt as well which we can run here?

not that I'm aware of.

**Review by adiantum — APPROVED** [review]
/lgtm
/approve

**Review by yannickstruyf3 — APPROVED** [review]

**Pull request merged**